packages/client/src/generation/TSClient/PrismaClient.ts

```diff
@@ -307,6 +307,14 @@ function runCommandRawDefinition(this: PrismaClientClass) {
   return ts.stringify(method, { indentLevel: 1, newLine: 'leading' })
 }
 
+function eventRegistrationMethodDeclaration(dataProxy: boolean) {
+  if (dataProxy) {
```

Comment:
Should we also generate the same on `getClientEngineType() === ClientEngineType.Library` ?

Reply:
Actually yes, I forgot that the engine type can't be switched without regenerating the client anymore!